api/grpcserver/v2alpha1/transaction.go

```diff
@@ -441,83 +434,31 @@ func toTxContents(rawTx []byte) (*spacemeshv2alpha1.TransactionContents,
 	txType := spacemeshv2alpha1.Transaction_TRANSACTION_TYPE_UNSPECIFIED
 
 	r := bytes.NewReader(rawTx)
-	method, template, txArgs, err := decodeTxArgs(scale.NewDecoder(r))
+	method, _, pubkey, err := decodeTxArgs(scale.NewDecoder(r))
 	if err != nil {
 		return res, txType, err
 	}
 
-	switch method {
-	case core.MethodSpawn:
-		switch *template {
-		case wallet.TemplateAddress:
-			args := txArgs.(*wallet.SpawnArguments)
-			res.Contents = &spacemeshv2alpha1.TransactionContents_SingleSigSpawn{
-				SingleSigSpawn: &spacemeshv2alpha1.ContentsSingleSigSpawn{
-					Pubkey: args.PublicKey.String(),
-				},
-			}
-			txType = spacemeshv2alpha1.Transaction_TRANSACTION_TYPE_SINGLE_SIG_SPAWN
-		case multisig.TemplateAddress:
-			args := txArgs.(*multisig.SpawnArguments)
-			contents := &spacemeshv2alpha1.TransactionContents_MultiSigSpawn{
-				MultiSigSpawn: &spacemeshv2alpha1.ContentsMultiSigSpawn{
-					Required: uint32(args.Required),
-				},
-			}
-			contents.MultiSigSpawn.Pubkey = make([]string, len(args.PublicKeys))
-			for i := range args.PublicKeys {
-				contents.MultiSigSpawn.Pubkey[i] = args.PublicKeys[i].String()
-			}
-			res.Contents = contents
-			txType = spacemeshv2alpha1.Transaction_TRANSACTION_TYPE_MULTI_SIG_SPAWN
-		case vesting.TemplateAddress:
-			args := txArgs.(*multisig.SpawnArguments)
-			contents := &spacemeshv2alpha1.TransactionContents_VestingSpawn{
-				VestingSpawn: &spacemeshv2alpha1.ContentsMultiSigSpawn{
-					Required: uint32(args.Required),
-				},
-			}
-			contents.VestingSpawn.Pubkey = make([]string, len(args.PublicKeys))
-			for i := range args.PublicKeys {
-				contents.VestingSpawn.Pubkey[i] = args.PublicKeys[i].String()
-			}
-			res.Contents = contents
-			txType = spacemeshv2alpha1.Transaction_TRANSACTION_TYPE_VESTING_SPAWN
-		case vault.TemplateAddress:
-			args := txArgs.(*vault.SpawnArguments)
-			res.Contents = &spacemeshv2alpha1.TransactionContents_VaultSpawn{
-				VaultSpawn: &spacemeshv2alpha1.ContentsVaultSpawn{
-					Owner:               args.Owner.String(),
-					TotalAmount:         args.TotalAmount,
-					InitialUnlockAmount: args.InitialUnlockAmount,
-					VestingStart:        args.VestingStart.Uint32(),
-					VestingEnd:          args.VestingEnd.Uint32(),
-				},
-			}
-			txType = spacemeshv2alpha1.Transaction_TRANSACTION_TYPE_VAULT_SPAWN
+	if spawnSelector, err := athcon.FromString("athexp_spawn"); err != nil {
+		return res, txType, fmt.Errorf("%w: failed to create spawn selector: %w", core.ErrInternal, err)
+	} else if spendSelector, err := athcon.FromString("athexp_spend"); err != nil {
+		return res, txType, fmt.Errorf("%w: failed to create spend selector: %w", core.ErrInternal, err)
+	} else if *method == spawnSelector {
```

Comment:
There is no need for `else` branches, as the previous branch returns on error. Consider:
```suggestion
	spawnSelector, err := athcon.FromString("athexp_spawn")
	if err != nil {
		return res, txType, fmt.Errorf("%w: failed to create spawn selector: %w", core.ErrInternal, err)
	}
	spendSelector, err := athcon.FromString("athexp_spend")
	if err != nil {
		return res, txType, fmt.Errorf("%w: failed to create spend selector: %w", core.ErrInternal, err)
	}
	switch *method {
	case  == spawnSelector:
	    // body
     	case  == spendSelector:
	    // body
```